T78750158 Support varying size input in numeric suite at 10/30/2020, 3:55:01 PM

DeepDriving opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #48056 Compare Weights FX Implementation
* **#47391 T78750158 Support varying size input in numeric suite at 10/30/2020, 3:55:01 PM**

Current Numeric Suite will fail if it's collecting for multiple inputs and each input is of not same size.  This fix adds support for varying size input in numeric suite.

Differential Revision: [D24662271](https://our.internmc.facebook.com/intern/diff/D24662271/)

Commit:
T78750158 Support varying size input in numeric suite at 10/30/2020, 3:55:01 PM

Current Numeric Suite will fail if it's collecting for multiple inputs and each input is of not same size.  This fix adds support for varying size input in numeric suite.

Differential Revision: [D24662271](https://our.internmc.facebook.com/intern/diff/D24662271/)

[ghstack-poisoned]

Commit:
Update on "T78750158 Support varying size input in numeric suite at 10/30/2020, 3:55:01 PM"


Current Numeric Suite will fail if it's collecting for multiple inputs and each input is of not same size.  This fix adds support for varying size input in numeric suite.

Differential Revision: [D24662271](https://our.internmc.facebook.com/intern/diff/D24662271/)

[ghstack-poisoned]

Commit:
Update on "T78750158 Support varying size input in numeric suite at 10/30/2020, 3:55:01 PM"


Current Numeric Suite will fail if it's collecting for multiple inputs and each input is of not same size.  This fix adds support for varying size input in numeric suite.

Differential Revision: [D24662271](https://our.internmc.facebook.com/intern/diff/D24662271/)

[ghstack-poisoned]

hx89 commented on test/quantization/test_numeric_suite.py:
```diff
@@ -435,7 +458,7 @@ def compare_and_validate_results(float_model, q_model, input, hidden):
             self.assertTrue(act_compare_dict.keys() == expected_act_compare_dict_keys)
             for k, v in act_compare_dict.items():
-                self.assertTrue(v["float"][0].shape == v["quantized"][0].shape)
+                self.assertTrue(len(v["float"]) == len(v["quantized"]))
```
May also check the shape of each element?

hx89 commented on test/quantization/test_numeric_suite.py:
```diff
@@ -447,3 +470,38 @@ def compare_and_validate_results(float_model, q_model, input, hidden):
+        logger.forward(b_float, b_quantized)
+
+        self.assertEqual(len(logger.stats["float"]), 2)
+        self.assertEqual(len(logger.stats["quantized"]), 2)
```
How about add check the contents are the same also?

Commit:
Update on "T78750158 Support varying size input in numeric suite at 10/30/2020, 3:55:01 PM"


Current Numeric Suite will fail if it's collecting for multiple inputs and each input is of not same size.  This fix adds support for varying size input in numeric suite.

Differential Revision: [D24662271](https://our.internmc.facebook.com/intern/diff/D24662271/)

[ghstack-poisoned]

Commit:
Update on "T78750158 Support varying size input in numeric suite at 10/30/2020, 3:55:01 PM"


Current Numeric Suite will fail if it's collecting for multiple inputs and each input is of not same size.  This fix adds support for varying size input in numeric suite.

Differential Revision: [D24662271](https://our.internmc.facebook.com/intern/diff/D24662271/)

[ghstack-poisoned]

Commit:
Update on "T78750158 Support varying size input in numeric suite at 10/30/2020, 3:55:01 PM"


Current Numeric Suite will fail if it's collecting for multiple inputs and each input is of not same size.  This fix adds support for varying size input in numeric suite.

Differential Revision: [D24662271](https://our.internmc.facebook.com/intern/diff/D24662271/)

[ghstack-poisoned]
